sample-app/src/components/VisualAutocomplete/VisualSearchBar.tsx

```diff
@@ -50,10 +56,23 @@ export default function VisualSearchBar({
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input) ?? '';
   const isLoading = useAnswersState(state => state.searchStatus.isLoading) ?? false;
-  const [executeQuery, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
+  const [executeQueryWithNearMeHandling, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
   const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(async () => {
     return answersActions.executeUniversalAutocomplete();
   });
+  const recentSearches = useMemo(() => {
+    return hideRecentSearches 
+      ? undefined
+      : new RecentSearches({ limit: recentSearchesLimit })
+  }, [recentSearchesLimit, hideRecentSearches]);
+
+  function executeQuery() {
+    if (!hideRecentSearches) {
+      let input = answersActions.state.query.input;
+      input && recentSearches?.setRecentSearch(input);
```

Comment:
Do we need the `?` after `recentSearches` here?

Reply:
typescript can't deduce that hideRecentSearches guard the instantiation of recentSearches and yell that recentSearches `is possibly 'undefined'`